.github/CONTRIBUTING.md

```diff
@@ -1,42 +1,251 @@
+
+#### How
+
+Always describe what's being fixed and not *how* it was fixed.
+
+Exemple of a **bad** PR title:
+
+```
+[FIX] Add Content-Type for public files with JWT
+```
+
+Exemple of a **good** PR title:
+
+```
+[FIX] Missing Content-Type header for public files with JWT
+```
+
+### `[IMPROVE]`
+
+#### When
+- When a change enhances a not buggy behavior. When in doubt if it's a Improve or Fix prefer to use as fix.
+
+#### How
+Always describe what's being improved and not *how* it was improved.
+
+Exemple of **good** PR title:
+
+```
+[IMPROVE] Displays Nothing found on admin sidebar when search returns nothing
+```
+
+### `[BREAK]`
+
+#### When
+- When the changes affect a working feature
+
+##### Back-End
+- When the API contract (data structure and endpoints) are limited, expanded as required or removed
+- When the business logic (permissions and roles) are limited, expanded (without migration) or removed
+
+##### Front-End
+- When the change limits (format, size, etc) or removes the ability of read or change the data (when the limitation was not caused by the back-end)
+
+### Second tag e.g. `[NEW][ENTERPRISE]`
+
+Use a second tag to group entries on the change log, we currently use it only for the Enterprise items but we are going to expand it's usage soon, please do not use it until we create a patter for it.
+
+### Minor Changes
+
+For those PRs that aren't important for the end user, we are working on a better pattern, but by now keep the same tags, but use them without the brackets and in camel case:
+
+```
+Fix: Missing Content-Type header for public files with JWT
+```
+
+All those PRs will be grouped under the `Minor changes` section which is collapsed, so users can expand it to check for those minor things but they are not visible directly on changelog.
+
+## Security Guidelines
+
+- Never expose unnecessary data to the APIs' responses
+- Always check for permissions or create new ones when need to expose sensitive data
+- Never provide new APIs without rate limiters
+- Always escape the user's input when rendering data
+- Always limit the user's input size on server side
+- Always execute the validations on the server side even when executing on the client side as well
+
+## Performance Guidelines
+
+- Inform the fields you want returned on a query for only those that are necessary
+- Limit the number of returner records to a reasonable value
+- Check if the query is using indexes, it it's not create new indexes
+- Prefer queues over long executions
+- Create new metrics to mesure things whenever possible
+- Cache data and returns whenever possible
+
+## Contributor License Agreement
+
+To have your contribution accepted you must sign our [Contributor License Agreement](https://cla-assistant.io/RocketChat/Rocket.Chat). In case you submit a Pull Request before sign the CLA GitHub will alert you with a new comment asking you to sign and will block the Pull Request from be merged by us.
 
```

Comment:
SUPER COOL !!!!    While I don't necessarily agree with all technical bits, **this** will NO QUESTION greatly enhance the contributors' experience and even our relationship with  technical community members.    THANKS!!!!